torchtune/modules/model_fusion/_early_fusion.py

```diff
@@ -130,23 +130,25 @@ def __init__(
         set_trainable_params(self, trainable_params)
 
     @staticmethod
-    def _state_dict_hook(module, state_dict, *args, **kwargs):
+    def _state_dict_hook(module, state_dict, prefix, *args, **kwargs):
         """
         Keep tok_embeddings inside of decoder state_dict
 
         [!Note] This update changes the order of the OrderedDict
         """
         for n, p in module.tok_embeddings.named_parameters():
-            state_dict[f"decoder.tok_embeddings.{n}"] = p
-            del state_dict[f"tok_embeddings.{n}"]
+            state_dict[f"{prefix}decoder.tok_embeddings.{n}"] = p
```

Comment:
prefix will include the period?

Reply:
Yep, see [here](https://github.com/pytorch/pytorch/blob/a56af029133ba0e87ed4c099e7e4efc40a24e20c/torch/nn/modules/module.py#L1892-L1894)